.circleci/cimodel/pytorch_build_definitions.py

```diff
@@ -0,0 +1,369 @@
+
+                if self.gpu_resource == "large":
+                    env_dict["MULTI_GPU"] = miniutils.quote("1")
+
+            d["resource_class"] = resource_class
+
+        return d
+
+    def gen_workflow_yaml_item(self, phase):
+
+        if self.is_xla or phase == "test":
+            val = OrderedDict()
+            if self.is_xla:
+                val["filters"] = {"branches": {"only": ["master"]}}
+
+            if phase == "test":
+                dependency_build = self.parent_build or self
+                val["requires"] = [dependency_build.gen_build_name("build")]
+
+            return {self.gen_build_name(phase): val}
+        else:
+            return self.gen_build_name(phase)
+
+
+# TODO This is a hack to special case some configs just for the workflow list
+class HiddenConf(object):
+    def __init__(self, name, parent_build=None):
+        self.name = name
+        self.parent_build = parent_build
+
+    def gen_workflow_yaml_item(self, phase):
+
+        val = OrderedDict()
+        dependency_build = self.parent_build
+        val["requires"] = [dependency_build.gen_build_name("build")]
+
+        return {self.gen_build_name(phase): val}
+
+    def gen_build_name(self, _):
+        return self.name
+
+
+xenial_parent_config = Conf(
+    "xenial",
+    ["py3"],
+    pyver="3.6",
+    cuda_version="8",
+    gpu_resource="medium",
+)
+
+
+# TODO This is a short-term hack until it is converted to recursive tree traversal
+xenial_dependent_configs = [
+    Conf("xenial",
+         ["py3", DockerHide("multigpu")],
+         pyver="3.6",
+         cuda_version="8",
+         restrict_phases=["test"],
+         gpu_resource="large",
+         parent_build=xenial_parent_config,
+         ),
+    Conf("xenial",
+         ["py3", DockerHide("NO_AVX2")],
+         pyver="3.6",
+         cuda_version="8",
+         restrict_phases=["test"],
+         gpu_resource="medium",
+         parent_build=xenial_parent_config,
+         ),
+    Conf("xenial",
+         ["py3", DockerHide("NO_AVX"), DockerHide("NO_AVX2")],
+         pyver="3.6",
+         cuda_version="8",
+         restrict_phases=["test"],
+         gpu_resource="medium",
+         parent_build=xenial_parent_config,
+         ),
+
+    HiddenConf("pytorch_short_perf_test_gpu", parent_build=xenial_parent_config),
+    HiddenConf("pytorch_doc_push", parent_build=xenial_parent_config),
+]
+
+
+xenial_parent_config.dependent_tests = xenial_dependent_configs
+
+
+# TODO this hierarchy is a work in progress
+CONFIG_TREE_DATA = [
+    ("trusty", [
+        ("py2.7.9", []),
+        ("py2.7", []),
+        ("py3.5", []),
+        ("py3.6", [
+            ("gcc4.8", []),
+            ("gcc5.4", [False, True]),
+            ("gcc7", []),
+        ]),
+        ("pynightly", []),
+    ]),
+    ("xenial", [
+        ("clang", [
+            ("X", [("py3", [])]),
+        ]),
+        ("cuda", [
+            ("8", [("3.6", [])]),
+            ("9", [
+                ("2.7", []),
```

Comment:
Note there are magic strings here https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh#L21 and https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh#L143 and https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh#L153